core/metrics-core-service/src/main/java/org/hawkular/metrics/core/jobs/DeleteTenant.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.jobs;
+
+import org.hawkular.metrics.core.service.MetricsService;
+import org.hawkular.metrics.model.Metric;
+import org.hawkular.metrics.model.MetricType;
+import org.hawkular.metrics.scheduler.api.JobDetails;
+import org.hawkular.rx.cassandra.driver.RxSession;
+import org.jboss.logging.Logger;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+
+import rx.Completable;
+import rx.Observable;
+import rx.functions.Func1;
+
+/**
+ * @author jsanda
+ */
+public class DeleteTenant implements Func1<JobDetails, Completable> {
+
+    private static Logger logger = Logger.getLogger(DeleteTenant.class);
+
+    public static final String JOB_NAME = "DELETE_TENANT";
+
+    private RxSession session;
+
+    private PreparedStatement deleteTenant;
+
+    private PreparedStatement deleteData;
+
+    private PreparedStatement deleteFromMetricsIndex;
+
+    private PreparedStatement deleteTag;
+
+    private PreparedStatement deleteRetentions;
+
+    private MetricsService metricsService;
+
+    public DeleteTenant(RxSession session, MetricsService metricsService) {
+        this.session = session;
+        this.metricsService = metricsService;
+        deleteTenant = session.getSession().prepare("DELETE FROM tenants WHERE id = ?");
+        deleteData = session.getSession().prepare(
+                "DELETE FROM data WHERE tenant_id = ? AND type = ? AND metric = ? AND dpart = 0");
+        deleteFromMetricsIndex = session.getSession().prepare(
+                "DELETE FROM metrics_idx WHERE tenant_id = ? AND type = ? AND metric = ?");
+        deleteTag = session.getSession().prepare("DELETE FROM metrics_tags_idx WHERE tenant_id = ? AND tname = ? " +
```

Comment:
Wouldn't it make more sense to just delete a partition? (tenant_id, tname). What's the benefit of giving tvalue, type and metric properties? We could get one row tombstone instead of multiple cell tombstones.